sympy/stats/stochastic_process_types.py

```diff
@@ -0,0 +1,25 @@
+from sympy import Indexed, Mul, Eq, sympify, S, Lambda
+from sympy.stats.frv import SingleFinitePSpace
+from sympy.stats.frv_types import BernoulliDistribution
+from sympy.stats.joint_rv import (JointDistributionHandmade, JointPSpace,
+    StochasticProcess)
+from sympy.functions.elementary.piecewise import Piecewise
+
+class BernoulliProcess(StochasticProcess):
```

Comment:
I had considered the StochasticProcess class as a collection of random variables, and any finite number of indexed elements from could be used to make a joint distribution. Each indexed element itself would a random variable.
Can you please explain as to how the StochasticProcess class could be made similar to JointRandomSymbol, i.e, how to seperate the distibution and the probability space?

Reply:
One way would be to represent it as a collection of distributions instead of random symbols.

Reply:
we extract the collection of probability spaces when needed. For example the expression `E(X[t], X[t-2])` would probably extract the probability spaces at `t` and `t-2`, thus creating the collection.

I suggest to keep the API as close as possible to the existing one, as it would avoid editing too much later.